tests/cut_plot_test.py

```diff
@@ -101,6 +101,19 @@ def test_update_legend_legends_shown(self):
         self.assertTrue(self.cut_plot._legends_visible[0])
         self.axes.legend.assert_called_with([line], ['some_label'], fontsize=ANY)
 
+    def test_update_legend_with_line_data(self):
+        line_data = [
+            {'shown': True, 'color': 'blue', 'style': '--', 'width': 1.5, 'marker': 'o', 'legend': 2, 'label': 'visible_line_data_label', 'error_bar': True},
+            {'shown': True, 'color': 'cyan', 'style': ':', 'width': 1.5, 'marker': 'o', 'legend': 0, 'label': 'non_visible_line_data_label', 'error_bar': True}
+        ]
+        mock_line = Line2D([], [])
+        another_mock_line = Line2D([], [])
+        self.axes.get_legend_handles_labels = MagicMock(return_value=([mock_line, another_mock_line], ['mock_label', 'another_mock_label']))
+
+        self.cut_plot.update_legend(line_data)
+        self.assertEqual(self.cut_plot._legends_visible, [2, 0])
+        self.axes.legend.assert_called_with([mock_line], ['visible_line_data_label'], fontsize=ANY)
+
```

Comment:
Added this unit test function to test labels that should be visible within the legends box